CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+
+Our framework has a few key concepts that you should be familiar with.
+
+### Slots
+
+The "Slot" pattern is an abstraction pattern used to compose higher order components. The pattern allows us to divide the props of one component into multiple "slots". An individual slot represents your inner component, in essence the actual entries that you will put in your render tree. Using the slot pattern lets you abstract away specifically what gets rendered in the tree and in advanced user scenarios allow you to switch out or flatten slots. For example, you might choose to replace whatever we put in the default "icon" slot for a button with a SVG, or a spinner. You could replace the root slot of a radio group with a focus trapzone if the radio button is a required part of the form. The props are assigned to whatever happens to get rendered for that slot. It is also built to help with style creation for the slots to help keep object references to the produced style objects consistent and use cached results where possible.
+
+One caveat is that if a component simply wraps a single native component, then it only has one slot.
+
+### Tokens
+
+FluentUI React Native (and eventually all of FluentUI) uses the design token system to handle styling / customization. This seems to be inspired by [Adobe Design Tokens](https://spectrum.adobe.com/page/design-tokens/). There's [documentation](https://github.com/microsoft/fluentui-react-native/blob/master/packages/framework/foundation-tokens/README.md) about this in the repo, but I'll try to give a simple overview.
+
+**Tokens** are things you set at design time, via theme, or via customizing the control. An example might be "brandColor", where each app has it's own color token it sets on all if it's controls
+
+**Props** are just normal properties that you can set on a JSX element. An example might be "isAnimating".
+
+Tokens can also be props (which is specified in the compose framework with the "TokensThatAreAlsoProps" field of the control)
+Tokens help us achieve simpler customization for complex higher order components, and also help with memoization (AKA rendering the minimal set of changes every time the component is re-rendered)
+
+## Creating a new component
+
+This section covers creating and adding a new component package to FluentUI React Native's monorepo. If you are instead working on an existing component and adding a native module, skip to the next two sections.
+
+1. Create a new directory in of these two locations, depending on your component:
+
+   - `fluentui-react-native/packages/components/<new-component>`
+   - `fluentui-react-native/packages/experimental/<new-component>`
+
+1. Copy the files from the top level of another component (A good example of this is the Shimmer). Make sure to empty the changelog file, and modify the name of the component where appropriate
+1. Update your `package.json` file. This defines the name / fields of the npm package that will be published for this component.
```

Comment:
This falls into the area where I'm not sure if we need to document that here of not, since readers should be familiar with package.json' and dependency management as that's documented publicly on the internet in many places. However, I guess I didn't write this guide with that idea in mind (don't re-document stuff that's already public info) so I'll add a sentence for that.